### PR TITLE
Updated domain exception

### PR DESCRIPTION
**Pull request by 0nary**
Add DomainException for ArgumentCannotBeADateInThePast.
Add DomainException for ArgumentMustBeEarlierThan with DateTime.
Add DomainException for ArgumentMustBeEarlierThan with DateTimeOffset.
Add DomainException for ArgumentMustBeLaterThan with DateTimeOffset.
Add DomainException for ArgumentMustBeComprisedBetweenIncluding with double.
Add DomainException for ArgumentMustBeComprisedBetweenIncluding with decimal.
Remove DomainException ArgumentMustBeLowerOrHigherThan.

### CODE REVIEW

File: src/Data/Neuroglia.Data/DomainException.cs [hunk 1]
```diff
@@ -353,6 +352,28 @@ public static DomainArgumentException ArgumentMustBeEarlierThan(string argumentN
             return new DomainArgumentException(StringExtensions.Format(LocalizedResources.exception_domain_argumentmustbeearlierthan, argumentName, time), argumentName);
         }
 
+        /// <summary>
+        /// Creates a new <see cref="DomainArgumentException"/> for an argument that must be lower than the specified time
```
Comment (cdavernas) [inline src/Data/Neuroglia.Data/DomainException.cs:356]: > ... must be lower than the specified time ...

I would replace with:

`must be lower than the specified date and time`

File: src/Data/Neuroglia.Data/DomainException.cs [hunk 2]
```diff
@@ -353,6 +352,28 @@ public static DomainArgumentException ArgumentMustBeEarlierThan(string argumentN
...
+        /// </summary>
+        /// <param name="argumentName">The argument's name</param>
+        /// <param name="date">The date the argument must be lower than</param>
+        /// <returns>A new <see cref="DomainArgumentException"/> for an argument that must be lower than the specified time</returns>
```
Comment (cdavernas) [inline src/Data/Neuroglia.Data/DomainException.cs:360]: Same comment than above

File: src/Data/Neuroglia.Data/DomainException.cs [hunk 3]
```diff
@@ -353,6 +352,28 @@ public static DomainArgumentException ArgumentMustBeEarlierThan(string argumentN
...
+        public static DomainArgumentException ArgumentMustBeEarlierThan(string argumentName, DateTime date)
+        {
+            return new DomainArgumentException(StringExtensions.Format(LocalizedResources.exception_domain_argumentmustbeearlierthan, argumentName, date), argumentName);
+        }
+
+        /// <summary>
+        /// Creates a new <see cref="DomainArgumentException"/> for an argument that must be lower than the specified time
```
Comment (cdavernas) [inline src/Data/Neuroglia.Data/DomainException.cs:367]: Same comment than above

File: src/Data/Neuroglia.Data/DomainException.cs [hunk 4]
```diff
@@ -353,6 +352,28 @@ public static DomainArgumentException ArgumentMustBeEarlierThan(string argumentN
...
+        /// </summary>
+        /// <param name="argumentName">The argument's name</param>
+        /// <param name="date">The date the argument must be lower than</param>
+        /// <returns>A new <see cref="DomainArgumentException"/> for an argument that must be lower than the specified time</returns>
```
Comment (cdavernas) [inline src/Data/Neuroglia.Data/DomainException.cs:371]: Same comment than above

File: src/Data/Neuroglia.Data/DomainException.cs [hunk 5]
```diff
@@ -375,6 +396,17 @@ public static DomainArgumentException ArgumentMustBeLaterThan(string argumentNam
             return new DomainArgumentException(StringExtensions.Format(LocalizedResources.exception_domain_argumentmustbelaterthan, argumentName, date), argumentName);
         }
 
+        /// <summary>
+        /// Creates a new <see cref="DomainArgumentException"/> for an argument that must be higher than the specified date
```
Comment (cdavernas) [inline src/Data/Neuroglia.Data/DomainException.cs:400]: > specified date

Please replace with:

`specified date and time`

File: src/Data/Neuroglia.Data/DomainException.cs [hunk 6]
```diff
@@ -375,6 +396,17 @@ public static DomainArgumentException ArgumentMustBeLaterThan(string argumentNam
...
+        /// </summary>
+        /// <param name="argumentName">The argument's name</param>
+        /// <param name="date">The date the argument must be higher than</param>
+        /// <returns>A new <see cref="DomainArgumentException"/> for an argument that must be higher than the specified date</returns>
```
Comment (cdavernas) [inline src/Data/Neuroglia.Data/DomainException.cs:404]: Same comment than above

File: src/Data/Neuroglia.Data/Neuroglia.Data.csproj [hunk 7]
```diff
@@ -40,4 +40,9 @@
   <ItemGroup>
     <Folder Include="Properties\" />
   </ItemGroup>
+  <ItemGroup>
```
Comment (cdavernas) [inline src/Data/Neuroglia.Data/Neuroglia.Data.csproj:43]: Why did you add LocalizedResources.resx to the csproj? Wasn't the case before

File: src/Data/Neuroglia.Data/Properties/LocalizedResources.fr.resx [hunk 8]
```diff
@@ -189,6 +189,9 @@
   <data name="exception_domain_argumentcannotbeadateinthefuture" xml:space="preserve">
     <value>L'argument '{argument}' ne peut pas être une date future</value>
   </data>
+  <data name="exception_domain_argumentcannotbeadateinthepast" xml:space="preserve">
+    <value>L'argument '{argument}' ne peut pas être une date passée</value>
```
Comment (cdavernas) [inline src/Data/Neuroglia.Data/Properties/LocalizedResources.fr.resx:193]: > L'argument '{argument}' ne peut pas être une date passée

Maybe best to say:

`L'argument '{argument}' ne peut pas être une date dans le passé`

@JBBianchi WDYT?

**Review by cdavernas — CHANGES_REQUESTED** [review]

File: src/Data/Neuroglia.Data/Properties/LocalizedResources.fr.resx (hunk 8 above)
Comment (0nary) [reply src/Data/Neuroglia.Data/Properties/LocalizedResources.fr.resx:193]: I think too, I write it to have the same wording as `exception_domain_argumentcannotbeadateinthefuture`.

File: src/Data/Neuroglia.Data/Neuroglia.Data.csproj (hunk 7 above)
Comment (0nary) [reply src/Data/Neuroglia.Data/Neuroglia.Data.csproj:43]: Apparently it wasn't.
I think this happens when I change its access modifier from `no code gen` to `public` to regenerate  the `LocalizedResources.Designer.cs` the switch it back to `no code gen`.
So I think this should be removed.

File: src/Data/Neuroglia.Data/DomainException.cs (hunk 1 above)
Comment (0nary) [reply src/Data/Neuroglia.Data/DomainException.cs:356]: Should I write : 

- `time` when the argument name is time and `date` when the argument name is date
- `date or time` to match booth case without changing the comment between overload
- `date and time` like you say
